### PR TITLE
Metrics collection and Multithreading PoC

### PR DESCRIPTION
**Pull request by DownstreamDataTeam**
# Description of change
Added metrics collection system (proprietary, for mambu)
Implemented a multithreading PoC, not used yet as it needs to be refined further
Moved refactored Tap to the root folder, renamed some other files

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_mambu/helpers/schemas/activities.json [hunk 1]
```diff
@@ -1,239 +1,85 @@
...
-        "null",
-        "string"
-      ]
-    },
-    "user_name": {
-      "type": [
-        "null",
-        "string"
-      ]
     }
```
Comment (dmosorast) [inline tap_mambu/helpers/schemas/activities.json:84]: There are a lot of fields removed in this file, and the fields in the underlying object are changed. Is this intentional? This is maybe not a hard breaking change, but will result in data not being populated for the fields that were removed.

File: tap_mambu/helpers/schemas/loan_repayments.json [hunk 2]
```diff
@@ -1,304 +1,129 @@
...
-    "tax_penalty_paid": {
-      "type": [
-        "null",
-        "string"
-      ],
-      "format": "singer.decimal"
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "encoded_key": {
```
Comment (dmosorast) [inline tap_mambu/helpers/schemas/loan_repayments.json:5]: Is this intentional? A lot of fields were removed here as well.

**Review by dmosorast — COMMENTED** [review]
Marking as comments for your review, please. Just want to make sure the schema changes are intentional and that we can handle them correctly (e.g., if we need to reach out to customers, or update docs).

**Review by dmosorast — APPROVED** [review]

**Comment by gonzalobottari** [comment]
hey Team! how can I test Multithread?

**Pull request merged**